gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/java/io/gravitee/rest/api/management/v2/rest/resource/api/ApiPlansResource.java

```diff
@@ -124,55 +127,58 @@ public Response createApiPlan(@Valid @NotNull NewPlanEntity newPlanEntity) {
 
         PlanEntity planEntity = planService.create(GraviteeContext.getExecutionContext(), newPlanEntity);
 
-        return Response.created(this.getLocationHeader(planEntity.getId())).entity(PlanMapper.INSTANCE.convert(planEntity)).build();
+        return Response.created(this.getLocationHeader(planEntity.getId())).entity(planMapper.convert(planEntity)).build();
     }
 
     @GET
     @Path("/{plan}")
     @Produces(MediaType.APPLICATION_JSON)
+    @Permissions({ @Permission(value = RolePermission.API_PLAN, acls = { RolePermissionAction.READ }) })
     public Response getApiPlan(@PathParam("plan") String plan) {
         final ExecutionContext executionContext = GraviteeContext.getExecutionContext();
-        if (
-            Visibility.PUBLIC.equals(apiSearchService.findById(executionContext, apiId).getVisibility()) ||
-            hasPermission(GraviteeContext.getExecutionContext(), RolePermission.API_PLAN, apiId, RolePermissionAction.READ)
-        ) {
-            PlanEntity planEntity = planService.findById(executionContext, plan);
-            if (!planEntity.getApiId().equals(apiId)) {
-                return Response
-                    .status(Response.Status.BAD_REQUEST)
-                    .entity("'plan' parameter does not correspond to the current API")
-                    .build();
-            }
+        final GenericPlanEntity planEntity = planSearchService.findById(executionContext, plan);
 
-            return Response.ok(PlanMapper.INSTANCE.convert(planEntity)).build();
+        if (!planEntity.getApiId().equals(apiId)) {
+            return Response.status(Response.Status.NOT_FOUND).entity(planNotFoundError(plan)).build();
         }
-        throw new ForbiddenAccessException();
+
+        return Response.ok(planMapper.convert(planEntity)).build();
     }
 
     @PUT
     @Path("/{plan}")
```

Comment:
The path param should follow the same format as others, and be suffixed by `Id`
```suggestion
    @Path("/{planId}")
```